src/BenchmarkDotNet/Toolchains/MonoWasm/WasmAppBuilder.cs

```diff
@@ -0,0 +1,201 @@
+// -*- indent-tabs-mode: nil -*-
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Reflection;
+using BenchmarkDotNet.Toolchains.MonoWasm;
+
+public class WasmAppBuilder
+{
+    // FIXME: Document
+
+    private readonly WasmSettings WasmSettings;
+
+    private readonly string TargetFrameworkMoniker;
+
+  
+
+    private Dictionary<string, Assembly> _assemblies;
+    private AssemblyResolver _resolver;
+
+
+    public WasmAppBuilder(WasmSettings wasmSettings, string targetFrameworkMoniker)
+    {
+        WasmSettings = wasmSettings;
+        TargetFrameworkMoniker = targetFrameworkMoniker;
+    }
+
+    public bool BuildApp (string programName, string projectRoot)
+    {
+        string appDir = Path.Combine(projectRoot, $"bin/{TargetFrameworkMoniker}/browser-wasm/publish");
+
+        string outputDir = Path.Combine(appDir, "output");
+
+        List<string> assemblySearchPaths = new List<string>
+                                  {
+                                    appDir
+                                  };
+
+        string mainAssemblyPath = Path.Combine(appDir, $"{programName}.dll");
+
+
+        if (!File.Exists(mainAssemblyPath))
+            throw new ArgumentException($"File MainAssembly='{mainAssemblyPath}' doesn't exist.");
+        if (!File.Exists(WasmSettings.WasmMainJS))
+            throw new ArgumentException($"File MainJS='{WasmSettings.WasmMainJS}' doesn't exist.");
+
+        var paths = new List<string>();
+        _assemblies = new Dictionary<string, Assembly>();
+
+        // Collect and load assemblies used by the app
+        foreach (string dir in assemblySearchPaths)
+        {
+            if (!Directory.Exists(dir))
+                throw new ArgumentException($"Directory '{dir}' doesn't exist or not a directory.");
+            paths.Add(dir);
+        }
+        _resolver = new AssemblyResolver(paths);
+        var mlc = new MetadataLoadContext(_resolver, "System.Private.CoreLib");
+
+        Assembly mainAssembly = mlc.LoadFromAssemblyPath (mainAssemblyPath);
+        Add(mlc, mainAssembly);
```

Comment:
you can remove all of the MetadataLoadContext and AssemblyResolver related logic, we don't need it for BDN.

it's enough to simply copy all *.dll files to `output/managed`, e.g. like this:

```suggestion
        _resolver = new AssemblyResolver(paths);
        var mlc = new MetadataLoadContext(_resolver, "System.Private.CoreLib");

        Assembly mainAssembly = mlc.LoadFromAssemblyPath (mainAssemblyPath);
        Add(mlc, mainAssembly);
        string[] _assemblies = Directory.GetFiles(appDir, "*.dll");
```